clients/src/main/java/org/apache/kafka/common/network/Selector.java

```diff
@@ -268,6 +280,13 @@ public void poll(long timeout) throws IOException {
         currentTimeNanos = endSelect;
         this.sensors.selectTime.record(endSelect - startSelect, time.milliseconds());
 
+        // Perform setup for instantly connected channels since they won't be included in selectedKeys()
+        if (connectableChannels.size() > 0)
+            for (KafkaChannel channel : connectableChannels) {
+                sensors.maybeRegisterConnectionMetrics(channel.id());
```

Comment:
Could we put this and the next line in a private method and reuse the method here and in line 299? Also, we may not have to duplicate the code if we handle immediately connected keys in the same while loop at line 293. See my other comment below.